File: bio-informatics/challenges/number-to-pattern.js
const { getNucleotideByIndex } = require('./nucleotide-utils');

const numberToPattern = (number, patternLength) => {
    let pattern = [];

    for (let i = 0; i < patternLength; i++) {
        const remainder = number % 4;
        const nucleotide = getNucleotideByIndex(remainder);
        pattern.push(nucleotide);
        number = Math.floor(number / 4);
    }

    return pattern.reverse().join('');
};

module.exports = numberToPattern;

File: bio-informatics/challenges/python/utils/number_to_pattern.py
import math
from utils import get_nucleotide_by_index

def number_to_pattern(number, patternLength):
  pattern = []

  for x in range(patternLength):
    remainder = number % 4
    nucleotide = get_nucleotide_by_index(remainder)
    pattern.insert(0, nucleotide)
    number = math.floor(number / 4)

  return "".join(pattern)